scripts/exrrfs_mpassit.sh

```diff
@@ -28,35 +29,45 @@ elif [[ "${MESH_NAME}" == "conus3km" ]]; then
   ref_lat=38.5
 fi
 #
-#  find cycle time
+# find init.nc
 #
-YYYYMMDDHH=${CDATE}
-YYYYMMDD=${CDATE:0:8}
-HH=${CDATE:8:2}
+n=0
+while [[ $n -le 12 ]]; do
+  CDATEic=$($NDATE -$((10#${n})) "${CDATE}")
+  PDYic=${CDATEic:0:8}
+  HHic=${CDATEic:8:2}
+  checkfile=${COMINrrfs}/${RUN}.${PDYic}/${HHic}/ic/${WGF}${MEMDIR}/init.nc
+  if [[ -s "${checkfile}" ]]; then
+     ln -sf "${checkfile}" .
+     break
+  else
+     n=$((n + 1))
+  fi
+done
 #
 # find forecst length for this cycle
 #
 fcst_length=${FCST_LENGTH:-1}
 fcst_len_hrs_cycles=${FCST_LEN_HRS_CYCLES:-"01 01"}
-fcst_len_hrs_thiscyc=$(${USHrrfs}/find_fcst_length.sh "${fcst_len_hrs_cycles}" "${cyc}" "${fcst_length}")
+fcst_len_hrs_thiscyc=$("${USHrrfs}/find_fcst_length.sh" "${fcst_len_hrs_cycles}" "${cyc}" "${fcst_length}")
 echo "forecast length for this cycle is ${fcst_len_hrs_thiscyc}"
 #
 # loop through forecast history files for this group
 #
 fhr_string=$( seq 0 $((10#${HISTORY_INTERVAL})) $((10#${fcst_len_hrs_thiscyc} )) )
-fhr_all=(${fhr_string})
+fhr_all=("${fhr_string}")
```

Comment:
@chunhuazhou I just found that Line 58 no longer works as expected after adding double quotes. Could you change this line as follows:

```
read -ra fhr_all <<< "${fhr_string}"  # convert fhr_string to an array
```

https://github.com/guoqing-noaa/rrfs-workflow/blob/5d357ffbbf6d1de31cfa73c85e5e51b5dc2c0517/scripts/exrrfs_mpassit.sh#L43

Reply:
Done. Thanks!